src/net.cpp

```diff
@@ -1239,8 +1239,18 @@ void CConnman::ThreadSocketHandler()
                             if (!pnode->ReceiveMsgBytes(pchBuf, nBytes, notify))
                                 pnode->CloseSocketDisconnect();
                             RecordBytesRecv(nBytes);
-                            if (notify)
+                            if (notify) {
+                                auto it(pnode->vRecvMsg.begin());
```

Comment:
vRecvMsg no longer needs cs_vRecvMsg starting here. It's only ever used on this thread. In reality, cs_vRecvMsg isn't needed at all, so it would probably be more clear to just remove it. copyStats is the only thing that accesses some of these vars, but it already operates without any locks, so the situation isn't changed from before.